bluebottle/accounting/utils.py

```diff
@@ -0,0 +1,293 @@
+from django.db.models import Sum, Count, Q
+from django.utils import timezone
+from datetime import datetime
+
+from .models import BankTransaction, RemoteDocdataPayment, RemoteDocdataPayout, BankTransactionCategory, BankAccount
+from decimal import Decimal
+
+
+def add_up(first, second):
+    """
+    Addition of two objects.
+
+    Distinguish addition of dictionaries, list and numbers.
+
+    return None when both are None
+    if one of the two is None, return the other one which is not None
+
+    if both are mydicts, add them up with the custom mydict.__add__
+    if both are mylists, add them up with the .add method
+
+    if both are convertable to a Decimal, return the addition of the numbers
+
+    in other cases, return None
+    """
+    if not first:
+        return second
+    elif not second:
+        return first
+    else:
+        if isinstance(first, mydict) and isinstance(second, mydict):
+            return first + second
+        elif isinstance(first, mylist) and isinstance(second, mylist):
+            return first.add(second)
+        else:
+            try:
+                return Decimal(str(first)) + Decimal(str(second))
+            except:
+                return None
+
+
+class mydict(dict):
+
+    def __add__(self, dict2):
+        if self == mydict():
+            return dict2
+
+        new_dict = mydict()
+
+        keys = self.keys()
+        if set(keys) != set(dict2.keys()):
+            return new_dict
+
+        for key in keys:
+            new_dict[key] = add_up(self[key], dict2[key])
+
+        return new_dict
+
+
+class mylist(list):
```

Comment:
List with a custom add method that uses the 'add_up' function to handle all kinds of values (another dict, a list, value, string, None, ..)

Return an empty mylist when the two given lists do not have the same length.

When both lists have the same length, construct a new MyList, with the same length as the two given lists, and with values that are calculated by the add_up function that can handle any kind of nested dicts, lists (of dicts) etc. (this same add_up function is used when each value in two MyDicts are added up)